kotlin-android/app/src/main/java/com/mongodb/tasktracker/model/ListAdapter.kt

```diff
@@ -87,6 +90,8 @@ internal class ListAdapter(data: OrderedRealmCollection<Task>) : RealmRecyclerVi
             val item = it.where<Task>().equalTo("_id", _id).findFirst()
             item?.status = _status
         }
+        // always close realms when you are done with them!
+        bgRealm.close()
```

Comment:
This doesn't quite look right either? You are closing before the async has a chance to run? (The question marks indicate I'm not entirely sure here. Best check with Christian.)

See the second answer here (re: kotlin): https://stackoverflow.com/questions/47287823/does-realm-executetransactionasync-close-the-instance?rq=1

Actually, if this is indeed on a background thread, do we really still need `executeTransactionAsync` instead of `executeTransaction`?

Also, check out [`try-with-resources`](https://realm.io/docs/java/latest/#closing-realms).

Reply:
I just confirmed the above with Christian.

Reply:
Switching this to a sync transaction to keep things simpler -- otherwise we need to close the realm in onSuccess/onError which is just a lot of boilerplate for a simple operation.

Reply:
I'd rather avoid the try/catch pattern because it's not a "normal" use case for try/finally (seeing try without catch is weird), and the more elegant "try with resources" syntax is only available in java, not kotlin, so I think what we've got now should work.